File: Contracts/Model/Product/UserInteract/Response/PurchaseDetailOptionResponse.cs
using Contracts.Model.Product.Response;

namespace Contracts.Model.Product.UserInteract.Response;

public class PurchaseDetailOptionResponse : BaseResponse
{
	public Guid DetailId { get; set; }
	public OptionCartResponse Option { get; set; }
	public int Quantity { get; set; }
	public Guid OptionId { get; set; }
	public double UnitPrice { get; set; }
	public Guid? AppliedId { get; set; }
	public bool IsReturned { get; set; }
}

File: ClientApp/src/app/core/interfaces/Product/UserInteract/Response/purchaseDetailOptionResponse.ts
import {BaseResponse} from "../../../baseResponse";
import {OptionCartResponse} from "../../Response/option-cart-response";
import {DiscountPreviewResponse} from "@core/interfaces/Product/Response/discountPreviewResponse";

export interface PurchaseDetailOptionResponse extends BaseResponse {
  detailId: string;
  image: string;
  quantity: number;
  optionId: string;
  option: OptionCartResponse;
  unitPrice: number;
  appliedId: string | null;
  isReturned: boolean;
  newQuantity: number;
  offerPrice: number;
  discount: DiscountPreviewResponse;
}
